modules/dnn/include/opencv2/dnn/all_layers.hpp

```diff
@@ -1166,6 +1166,15 @@ CV__DNN_INLINE_NS_BEGIN
         static Ptr<ExpandLayer> create(const LayerParams &params);
     };
 
+    class CV_EXPORTS AttentionLayer : public Layer {
+     public:
+        int num_heads;
+        std::vector<int> qkv_hidden_sizes;
+        float scale;
```

Comment:
Can be moved to implementation? To avoid future ABI changes

Reply:
Yes, will do. We have had this issue, after all.

Reply:
Done.